go/Connection/TypeDBDatabaseImpl.go

```diff
@@ -0,0 +1,87 @@
+package Connection
+
+import (
+	"typedb_driver/go/api/database"
+	"typedb_driver/go_wrapper"
+)
+
+type TypeDBDatabaseImpl struct {
```

Comment:
What's the naming convention for `go` and interfaces: interface `IName` + impl `Name` or interface `Name` + impl `NameImpl`?

Reply:
Just checking.